Connection/RESTConnector.cs

```diff
@@ -262,11 +262,16 @@ public Request()
         /// <param name="function">The name of the function.</param>
         /// <returns>Returns a RESTConnector object or null on error.</returns>
         ///
-        public static RESTConnector GetConnector(Authenticator authenticator, string function)
+        public static RESTConnector GetConnector(Authenticator authenticator, string function, string url)
         {
+            if (Utility.HasBadFirstOrLastCharacter(url))
```

Comment:
We also need to make sure it's not null/nil or "".   In other words, we need to do what we can to ensure that by the time we want to use the serviceUrl value to connect to the server, it is as valid as we can determine :)

Reply:
Fair enough, I didn't add it because we are setting it to default URL if it's null or empty. But probably better to keep the check anyways. Let me update.

Reply:
yeah, the reason why this is important to validate against null/"" here is that after constructing the service, the app could set the URL to null or "".
In the parameterized url scenario (aka full iam) there won't be a default base url at all.